### ISSUE TITLE
Use container for get rule instances

### ISSUE THREAD

**Issue by vjik** [yiisoft/rbac#107]
• Remove `RuleFactoryInterface` and him implementation.
• Get rules via PSR-11 DI container.
• Remove `Manager::updateRule()`.
• `Manager::addRule()` and `Manager::removeRule()` should accept only rule names.
• RBAC stroages should store only rule names without data and not use serialization.

This way to work with rules looks more simpler and clearer.

**Comment by vjik** [comment]
Also it will allow don't store rules in roles storage.

**Comment by vjik** [comment]
Decided with team make internal container with rules that should use PSR container for resolve dependencies of rules.

**Comment by samdark** [comment]
Yes, isolation is needed. Mixing rules an services is not a good idea.
